### PR TITLE
[bitnami/postgresql-ha] Fix password when using existingSecret

### LINKED ISSUE #4655: [postgresql-ha] postgresql-postgres-password from existingSecret is ignored if postgresql.postgresPassword is not set
**Expectation:**
I can set the the postgres user's password by setting the key `postgresql-postgres-password` inside a secret referenced by `existingSecret`.

**Reality:**
I also need to specify any value in  postgresql.postgresPassword (default is nil), so that the secret gets referenced inside the envs, see
https://github.com/bitnami/charts/blob/6257aa55ca352a9aecac99cdf8491d4cccfa4102/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml#L120

### PR DESCRIPTION
**Pull request by miguelaeh**
**Description of the change**

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #4655

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by kirschem — COMMENTED** [review]
Looks valid to me, however I don't have time atm to do a test run with values.

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 1]
```diff
@@ -122,7 +122,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres")) }}
+            {{- if or .Values.postgresql.existingSecret (and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres"))) }}
```
Comment (juan131) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:125]: This assumes that you're using a user different from "postgres" when an existing secret is used to store the credentials.

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 2]
```diff
@@ -122,7 +122,7 @@ spec:
               value: {{ .Values.persistence.mountPath | quote }}
             - name: PGDATA
               value: {{ printf "%s/%s" .Values.persistence.mountPath "data" | quote }}
-            {{- if and .Values.postgresql.postgresPassword (not (eq .Values.postgresql.username "postgres")) }}
+            {{- if and (or .Values.postgresql.existingSecret .Values.postgresql.postgresPassword) (not (eq .Values.postgresql.username "postgres")) }}
```
Comment (juan131) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:125]: Better.. But we still have a problem.. This is not considered "global" values.
```suggestion
            {{- if and (or (include "postgresql-ha.postgresqlCreateSecret" .) (include "postgresql-ha.postgresqlPostgresPassword" .)) (not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres")) }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**